### PR TITLE
GHC 8.6 compatibility

### PR DESCRIPTION
**Pull request by quasicomputational**
This is based on #179.

It's not actually going to pass CI just yet because `th-expand-syns` needs a new release (but DanielSchuessler/th-expand-syns#11 has been merged, so hopefully that'll happen any moment now).

Once this is merged, a release would be very handy, please.

### CODE REVIEW

**Comment by DanielSchuessler** [comment]
@quasicomputational: I did as @hvr suggested and added a metadata revision on Hackage; hope that works for you (might have to retrigger a Travis build).

File: serialise/bench/versus/Macro/DeepSeq.hs [hunk 1]
```diff
@@ -1,9 +1,13 @@
+{-# LANGUAGE CPP #-}
 {-# OPTIONS_GHC -fno-warn-orphans #-}
 module Macro.DeepSeq where
 
 import Macro.Types
 
 import Control.DeepSeq
+#if MIN_VERSION_base(4,9,0)
```
Comment (hvr) [inline serialise/bench/versus/Macro/DeepSeq.hs]: shouldn't this rather be `MIN_VERSION_deepseq`? the `rnf1`/`rnf2` availability is not a proper function of the `base` version

**Comment by hvr** [comment]
@bgamari @thoughtpolice while there's already build-plans for GHC 8.6 (see https://matrix.hackage.haskell.org/package/serialise) the relaxations for the library components requested here can be safely performed as metadata revisions (it's mostly about accepting `containers-0.6`)

if you don't have time, do you want me to perform those revisions to expedite things and unblock @quasicomputational ?

**Comment by quasicomputational** [comment]
FWIW, revisions wouldn't really help me for a few reasons (I'm already using HEAD, and the GHC 8.6 upgrade plan is blocked elsewhere for me). It might benefit other users, though.

**Comment by bgamari** [comment]
Thanks for doing this!

**Pull request merged**